autoload/go/job.vim

```diff
@@ -254,6 +267,19 @@ function! go#job#Start(cmd, options)
     let l:options = s:neooptions(l:options)
   endif
 
+  " Verify that the working directory for the job actually exists. Return
+  " early if the directory does not exist. This helps avoid errors when
+  " working with plugins that use virtual files that don't actually exist on
+  " the file system.
+  let dir = expand("%:p:h")
+  if has_key(l:options, 'cwd')
+    if !isdirectory(l:options.cwd)
```

Comment:
This can be simplified to `if has_key(l:options, 'cwd')  && !isdirectory(l:options.cwd)` I guess ?